src/app/core/services/user_service.py

```diff
@@ -382,15 +561,68 @@ def _get_user_email(user_id:int, session: ScopedSession):
 			NotFoundError: User not found
 			Exception: Database error
 	"""
+	query = select(User).where(User.user_id == user_id)
 	try:
-		user = session.get(User, user_id)
+		user: User= session.execute(query).scalar_one()
+	except exc.NoResultFound as e:
+		logging.error('Failed to find user')
+		logging.exception(e)
+		raise NotFoundError(f'Could not find user with id: {user_id}')
 	except Exception as e:
 		logging.error(f'Failed to retrieve user')
 		logging.exception(e)
 		raise e
+	
+	return user.email
 
-	return user
+def _get_user_id(email:str, session: ScopedSession):
+	query = select(User).where(User.email == email)
```

Comment:
if you are just getting a single value its probably more efficient to just get the column